crates/parser2/src/parser/expr.rs

```diff
@@ -184,15 +198,28 @@ impl super::Parse for UnExprScope {
     }
 }
 
-define_scope! { BinExprScope, BinExpr, Inheritance }
+define_scope! { BinExprScope, BinExpr,Inheritance  }
 impl super::Parse for BinExprScope {
     fn parse<S: TokenStream>(&mut self, parser: &mut Parser<S>) {
         parser.set_newline_as_trivia(false);
 
         let (_, rbp) = infix_binding_power(parser).unwrap();
-        bump_bin_op(parser);
-
-        parse_expr_with_min_bp(parser, rbp, true);
+        if is_aug(parser) {
+            self.set_kind(SyntaxKind::AugAssignExpr);
+            bump_aug_assign_op(parser);
+            let (_, rbp) = infix_binding_power(parser).unwrap();
+            parser.bump_or_recover(SyntaxKind::Eq, "expected `=`", None);
+            parse_expr_with_min_bp(parser, rbp, true);
+        } else if is_asn(parser) {
+            self.set_kind(SyntaxKind::AssignExpr);
+            parser.set_newline_as_trivia(false);
+            let (_, rbp) = infix_binding_power(parser).unwrap();
+            parser.bump_or_recover(SyntaxKind::Eq, "expected `=`", None);
+            parse_expr_with_min_bp(parser, rbp, true);
+        } else {
+            bump_bin_op(parser);
+            parse_expr_with_min_bp(parser, rbp, false);
+        }
```

Comment:
```suggestion
        if is_aug(parser) {
            self.set_kind(SyntaxKind::AugAssignExpr);
            bump_aug_assign_op(parser);
            parser.bump_or_recover(SyntaxKind::Eq, "expected `=`", None);
            parse_expr_with_min_bp(parser, rbp, true);
        } else if is_asn(parser) {
            self.set_kind(SyntaxKind::AssignExpr);
            parser.set_newline_as_trivia(false);
            parser.bump_or_recover(SyntaxKind::Eq, "expected `=`", None);
            parse_expr_with_min_bp(parser, rbp, true);
        } else {
            bump_bin_op(parser);
            parse_expr_with_min_bp(parser, rbp, false);
        }
```

It's not necessary to recalculate `rbp`, also this would fix a bug that is introduced in `if is_aug` block(the `infix_binding_power` returned a wrong binding power).